libgalois/include/katana/PropertyGraph.h

```diff
@@ -646,36 +660,59 @@ class KATANA_EXPORT PropertyGraph {
   PropertyView node_property_view() {
     return PropertyView{
         .g = this,
+        .const_g = this,
         .schema_fn = &PropertyGraph::node_schema,
-        .property_fn = &PropertyGraph::GetNodeProperty,
-        .properties_fn = &PropertyGraph::node_properties,
+        .property_fn_int = &PropertyGraph::GetNodeProperty,
+        .property_fn_str = &PropertyGraph::GetNodeProperty,
+        .property_num_fn = &PropertyGraph::GetNumNodeProperties,
         .add_properties_fn = &PropertyGraph::AddNodeProperties,
         .upsert_properties_fn = &PropertyGraph::UpsertNodeProperties,
         .remove_property_int = &PropertyGraph::RemoveNodeProperty,
         .remove_property_str = &PropertyGraph::RemoveNodeProperty,
```

Comment:
Why are we using function pointers here instead of a simple subclass? It would eliminate the need for the explicit implementations of all the silly little methods that really just implement a vtable.

Reply:
Donald did the PropertyView class initially.  Do you mean a nested class or subclass?  `PropertyView` isn't a subclass of any class.

Reply:
What I mean is that you could have a pure virtual base class and then a few subclasses implementing that. Not that there is currently such a structure. That said, I just realized why this approach was probably chosen. The methods are mostly just methods that already exist on `PropertyGraph`, so this MIGHT be slightly simpler to implement.

I'll move on. :-)